src/core/src/op/reshape.cpp

```diff
@@ -94,25 +91,20 @@ bool Reshape::evaluate_upper(ov::TensorVector& output_values) const {
 }
 
 bool Reshape::evaluate_label(TensorLabelVector& output_labels) const {
-    if (!get_input_tensor(1).has_and_set_bound())
-        return false;
-    OPENVINO_SUPPRESS_DEPRECATED_START
-    return ov::default_label_evaluator(this, output_labels);
-    OPENVINO_SUPPRESS_DEPRECATED_END
+    return get_input_tensor(1).has_and_set_bound() && default_label_evaluator(this, {0}, output_labels);
 }
 
 bool Reshape::constant_fold(OutputVector& output_values, const OutputVector& inputs_values) {
     if (get_output_partial_shape(0).is_dynamic() || is_const_fold_disabled()) {
         return false;
     }
 
-    const auto& shape = get_output_shape(0);
-
-    if (auto data_const = std::dynamic_pointer_cast<op::v0::Constant>(inputs_values[0].get_node_shared_ptr())) {
-        output_values[0] = std::make_shared<op::v0::Constant>(*data_const, shape);
+    if (auto data_const = std::dynamic_pointer_cast<v0::Constant>(inputs_values[0].get_node_shared_ptr())) {
+        output_values[0] = std::make_shared<v0::Constant>(*data_const, get_output_shape(0));
         return true;
+    } else {
+        return false;
```

Comment:
Any reason for moving this `return false` to the `else`?

Reply:
Yes, it gives smaller binary size and logically is more correct as these two block are explicitly excluding.